opm/simulators/utils/ParallelRestart.hpp

```diff
@@ -221,32 +164,28 @@ template<class T1, class T2, class H, class P, class A>
 void unpack(std::unordered_map<T1,T2,H,P,A>& data, std::vector<char>& buffer, int& position,
             Dune::MPIHelper::MPICommunicator comm);
 
-void unpack(data::Rates& data, std::vector<char>& buffer, int& position,
-            Dune::MPIHelper::MPICommunicator comm);
-
-void unpack(data::Connection& data, std::vector<char>& buffer, int& position,
-            Dune::MPIHelper::MPICommunicator comm);
-
-void unpack(data::Segment& data,std::vector<char>& buffer, int& position,
-            Dune::MPIHelper::MPICommunicator comm);
-
-void unpack(data::Well& data, std::vector<char>& buffer, int& position,
-            Dune::MPIHelper::MPICommunicator comm);
-
-void unpack(RestartKey& data, std::vector<char>& buffer, int& position,
-            Dune::MPIHelper::MPICommunicator comm);
-
-void unpack(data::CellData& data, std::vector<char>& buffer, int& position,
-            Dune::MPIHelper::MPICommunicator comm);
-
-void unpack(data::Solution& data, std::vector<char>& buffer, int& position,
-            Dune::MPIHelper::MPICommunicator comm);
-
-void unpack(data::WellRates& data, std::vector<char>& buffer, int& position,
+void unpack(char* str, std::size_t length, std::vector<char>& buffer, int& position,
             Dune::MPIHelper::MPICommunicator comm);
 
-void unpack(RestartValue& data, std::vector<char>& buffer, int& position,
-            Dune::MPIHelper::MPICommunicator comm);
+/// prototypes for complex types
+
+#define ADD_PACK_PROTOTYPES(T) \
+  std::size_t packSize(const T& data, Dune::MPIHelper::MPICommunicator comm); \
+  void pack(const T& data, std::vector<char>& buffer, int& position, \
+          Dune::MPIHelper::MPICommunicator comm); \
+  void unpack(T& data, std::vector<char>& buffer, int& position, \
+              Dune::MPIHelper::MPICommunicator comm);
+
+ADD_PACK_PROTOTYPES(data::CellData)
+ADD_PACK_PROTOTYPES(data::Connection)
+ADD_PACK_PROTOTYPES(data::Rates)
+ADD_PACK_PROTOTYPES(data::Segment)
+ADD_PACK_PROTOTYPES(data::Solution)
+ADD_PACK_PROTOTYPES(data::Well)
+ADD_PACK_PROTOTYPES(data::WellRates)
+ADD_PACK_PROTOTYPES(RestartKey)
+ADD_PACK_PROTOTYPES(RestartValue)
+ADD_PACK_PROTOTYPES(std::string)
```

Comment:
Would have been easier if the old ordering would have been used. Seems like at least `const char*` is missing.

Otherwise I am neutral.

Reply:
const char* has another signature so it's not included.

Reply:
Indeed. 
From the diff it seemes as the lines 
```
void unpack(char* str, std::size_t length, std::vector<char>& buffer, int& position,
            Dune::MPIHelper::MPICommunicator comm);
```
were removed. But they were added again below . Did not see that due the reordering